syntax = "proto2";

package skunk;

message Argument {
   enum ArgumentType {
    eightByteArg = 0;
    stringArg = 1;
  }

  required ArgumentType type = 1;

  optional int64 argInt8b = 2;
  optional string argString = 3;
}

message ReturnValue {
  enum CallStatus {
    Success = 0;
    FunctionDoesntExist = 1;
    BadProtobufMessage = 2;
  }
    required CallStatus status = 1;
    optional int64 ret64 = 2;
    optional string retString = 3;
}

message FunctionCall {
    enum ReturnType {
      eightByte = 0;
    }
    required ReturnType returnType = 1;

    required string name = 2;
    required uint32 numberOfArguments = 3;
    optional Argument arg1 = 4;
    optional Argument arg2 = 5;
}